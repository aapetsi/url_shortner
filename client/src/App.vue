<template>
  <div>
    <nav>
      <a href="/">App</a>
      <a href="/login">Login</a>
      <a href="/register">Register</a>
      <a href="/logout">Logout</a>
    </nav>

    <router-view></router-view>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.extend({

})
</script>

<style scoped>
  nav {
    display: flex;
    justify-content: space-between;
  }
</style>