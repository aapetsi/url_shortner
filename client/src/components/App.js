import React, { useState, useEffect } from 'react'
import axios from 'axios'
import UrlList from './UrlList'
import UrlForm from './UrlForm'

const App = () => {
  const [urls, setUrls] = useState([])

  useEffect(() => {}, [urls])
  return (
    <div>
      <h1>URL Shortener</h1>
      <UrlForm />
      <UrlList urls={urls} />
    </div>
  )
}

export default App
