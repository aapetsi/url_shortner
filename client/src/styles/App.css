.container {
  display: flex;
  flex-direction: column;
  text-align: center;
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande',
    'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
}

.link {
  width: 30%;
  color: blue;
  text-decoration: underline;
  cursor: pointer;
  margin: 0 auto;
  padding-top: 15px;
  padding-bottom: 15px;
  height: 32px;
}

.delete-all {
  margin: 0 auto;
  background: #dc3545;
  height: 32px;
}

.delete-all[disabled]:hover {
  cursor: not-allowed;
}

.delete-one {
  width: 100%;
  height: 32px;
  background: #dc3545;
}

.copy-clipboard {
  display: inline-block;
  background: green;
  height: 30px;
  margin-left: 10px;
  width: 60px;
}

table {
  margin: 0 auto;
  width: 60%;
}

thead {
  background-color: blue;
  color: white;
  height: 32px;
}

tr:nth-child(even) {
  background-color: #f2f2f2;
}

form {
  display: flex;
  justify-content: center;
  margin-bottom: 50px;
}

input {
  width: 400px;
  height: 35px;
  padding: 5px;
  font-size: 18px;
  border: 1px solid #e3e3e3;
  border-radius: 2px;
  margin: auto 20px;
}

button {
  width: 90px;
  height: 47px;
  border-radius: 5px;
  cursor: pointer;
  color: white;
  background-color: blue;
  font-weight: bold;
  text-transform: uppercase;
}

button:hover {
  background: #556476;
}

@media only screen and (max-width: 540px) {
  form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    align-content: center;
    margin: auto;
  }

  input {
    width: 70%;
  }

  button {
    width: 73%;
    margin-top: 15px;
  }

  table {
    width: 90%;
  }
}
