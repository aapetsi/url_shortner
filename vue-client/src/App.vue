<template>
  <div id="app">
    <h1>URL Shortener</h1>
    <UrlForm v-bind:shortenUrl="shortenUrl" />
    <UrlList v-bind:urls="urls" v-bind:error="error" v-bind:handleDelete="handleDelete" />
    <button @click="handleDeleteAll" :disabled="urls.length === 0">Delete all</button>
  </div>
</template>

<script>
import UrlList from './components/UrlList.vue'
import UrlForm from './components/UrlForm.vue'
import Axios from 'axios'

export default {
  name: 'App',
  components: {
    UrlForm,
    UrlList
  },
  data: function() {
    return {
      urls: [],
      error: ''
    }
  },
  methods: {
    shortenUrl: async function(originalUrl) {
      try {
        const res = await Axios.post('http://localhost:3000/api/url/createShortLink', { originalUrl })
        this.urls = [...this.urls, res.data]
        this.error = ''
      } catch (error) {
        this.error = error.response.data.error
      }
    },

    handleDeleteAll: async function() {
      const ans = confirm('Are you sure you want to delete all urls')
      try {
        if (ans) {
          const res = await Axios.delete('http://localhost:3000/api/url/all', {data: ''})
          alert(res.data.message)
          this.urls = []
        }
      } catch (error) {
        this.error = 'There was a problem deleting all urls'
      }
    },

    handleDelete: async function(id) {
      try {
        const ans = confirm('Are you sure want to delete this url')
        if (ans) {
          await Axios.delete(`http://localhost:3000/api/url/one/${id}`, {data: ''})
          this.urls = this.urls.filter(url => url._id !== id)
        }
      } catch (error) {
        this.error = 'There was a problem deleting the url'
      }
      console.log('handleDelete')
    }
  },
  mounted: async function() {
    try {
      const res = await Axios.get('http://localhost:3000/api/url/get_urls')
      this.urls = res.data
    } catch (error) {
      this.error = 'Oops there seems to be a problem fetching your saved urls'
    }
  },
}
</script>

<style>

</style>
