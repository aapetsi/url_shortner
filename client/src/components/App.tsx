import React, { useState, useEffect } from 'react'
import Axios from 'axios'
import UrlList from './UrlList'
import UrlForm from './UrlForm'
import '../styles/App.css'

const App = () => {
  const [urls, setUrls] = useState<Array<{_id: string, shortUrl: string, originalUrl: string, shortUrlHash: string}>>([])
  const [error, setError] = useState<string>('')

  const shortenUrl = (originalUrl: string) => {
    Axios
      .post('http://localhost:3000/api/url/createShortLink', { originalUrl })
      .then((res) => {
        setUrls([...urls, res.data])
        setError('')
      })
      .catch((err) => {
        setError('There was a problem saving your URL to the database. Check your connection and try again')
      })
  }

  const fetchUrls = async () : Promise<void> => {
    const res = await Axios.get('http://localhost:3000/api/url/get_urls')
    if (res.status === 200 || res.statusText === '  OK') {
      setUrls([...urls, res.data])
    } else {
      setError('Oops there seems to be a problem with the network.\n Check your connection and try again')
    }
  }

  useEffect(() => { 
    // const res = fetchUrls()
    Axios
      .get('http://localhost:3000/api/url/get_urls')
      .then((res) => {
        setUrls([...urls, ...res.data])
        setError('')
      })
      .catch((err) => {
        setError('Oops there seems to be a problem with the network. Check your connection and try again')
      })
  }, [])
  
  return (
    <div className='container'>
      <h1>URL Shortener</h1>
      <UrlForm shortenUrl={shortenUrl} />
      <UrlList urls={urls} error={error} />
    </div>
  )
}

export default App