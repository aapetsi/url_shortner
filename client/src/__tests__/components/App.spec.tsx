import React from 'react'
import App from '../../components/App'
import {render, fireEvent, waitForElement} from '@testing-library/react'

describe('<App />', () => {
  test('should display the app', () => {
    expect(1).toBe(1)
  })
})