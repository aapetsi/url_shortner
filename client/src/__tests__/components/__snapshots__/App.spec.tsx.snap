// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render App 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1
      data-testid="title"
    >
      URL Shortener
    </h1>
    <div>
      <form
        data-testid="url-form"
      >
        <input
          data-testid="url"
          name="text"
          placeholder="https://example.com"
          required=""
          type="text"
          value=""
        />
        <button
          data-testid="submit"
        >
          Shorten
        </button>
      </form>
    </div>
    <div>
      <p
        data-testid="no-urls"
      >
        Start shortening your urls
      </p>
      <p
        data-testid="error"
      >
        Oops there seems to be a problem with the network. Check your connection and try again
      </p>
      <table>
        <thead>
          <tr>
            <td>
              Original Url
            </td>
            <td>
              Shortened Url
            </td>
            <td>
              Action
            </td>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <button
      class="delete-all"
      disabled=""
    >
      Delete all
    </button>
  </div>
</DocumentFragment>
`;
